woocommerce/woocommerce-gateway-stripe
# Set the subscription payment method to SEPA when purchasing using a sepa token method (eg iDEAL)

Fixes #3069

## Changes proposed in this Pull Request:

When a customer purchases a subscription, the subscription's payment method is generally inherited from the order. 

In some cases, like iDEAL, Bancontact, they use a different processing payment method for subscription payments - ie SEPA. 

Setting the subscription's payment method to SEPA typically happens once the customer returns to the site and is handled by the `process_upe_redirect_payment()` function. However, if the UPE redirect doesn't process the request, the subscription will incorrectly be left set to iDEAL and that will cause issues when the subscription goes to renew as iDEAL isn't a subscription processing payment method. 

This PR fixes that by setting the subscription's payment method to SEPA as the intent is being processed and before the customer leaves the site.  

## Testing instructions

1. Set up WooCommerce Subscriptions 
    - Install and activate WooCommerce Subscriptions
    - Create a simple subscription product
1. Set your site's Stripe account to a EU based account. 
3. Set your store's currency to Euros. 
4. Enable the iDEAL payment method. 
5. Add the subscription product to your cart.
6. Select iDEAL on checkout.
7. When redirected to the bank site leave that tab open.
8. View the subscriptions in your Admin dashboard: 
    - On `develop` you'll notice they are set to iDEAL
    - Repeat the steps on this branch and the subscription will be set to SEPA

| `develop` | This branch |
|--------|--------|
| <img width="822" alt="Screenshot 2024-05-08 at 4 09 20 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/22fee3af-cd51-48a1-b6e8-2d9511509ddc"> | <img width="871" alt="Screenshot 2024-05-08 at 4 06 26 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/73b6b0f7-a2e8-4604-9be4-5055152e05e9"> |

---


**Post merge**
